docs/snippets/src/main.c

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! [include]
+#include <openvino/c/openvino.h>
+//! [include]
+
+int main() {
+
+//! [part1]
+ov_core_t* core = NULL;
+ov_core_create(&core);
+//! [part1]
+
+const char* device_name = "CPU";
+{
+//! [part2_1]
+ov_compiled_model_t* compiled_model = NULL;
+ov_core_compile_model_from_file(core, "model.xml", device_name, 0, &compiled_model);
+//! [part2_1]
+}
+{
+//! [part2_2]
+ov_compiled_model_t* compiled_model = NULL;
+ov_core_compile_model_from_file(core, "model.onnx", device_name, 0, &compiled_model);
+//! [part2_2]
+}
+{
+//! [part2_3]
+ov_compiled_model_t* compiled_model = NULL;
+ov_core_compile_model_from_file(core, "model.pdmodel", device_name, 0, &compiled_model);
+//! [part2_3]
+}
+
+//! [part2_4]
+// Construct a model
+ov_model_t* model = NULL; // need to free by ov_model_free(model)
+ov_core_read_model(core, "model.xml", NULL, &model);
+ov_compiled_model_t* compiled_model = NULL;
+ov_core_compile_model(core, model, device_name, 0, &compiled_model);
+//! [part2_4]
+
+
+//! [part3]
+ov_infer_request_t* infer_request = NULL;
+ov_compiled_model_create_infer_request(compiled_model, &infer_request);
+//! [part3]
+
+ov_shape_t input_shape = {0, NULL};
```

Comment:
Can we use `get_shape()` to have correct data?

Reply:
using get input shape now

Reply:
@zhaixuejun1993 As I see we still have incorrect input_shape :)

Reply:
@ilyachur  Sorry, it's my fault, the change was reverted. Now, I had added back. Pls review.